### ISSUE TITLE
[BUG] - Smelter Mark 2 Missing from Production Tab and Incompatible with Blueprints

### ISSUE THREAD

**Issue by Flam1ngDem0n** [kremnev8/DSP-Mods#63]
Bug Description:
Smelter Mark 2 does not appear in the "Production Tab", it also does not work with the Blueprint system in the game when using a "Saved Blueprint". When I simply use the "Create Blueprint" option when building setup and then click "Use Now" the error does not show up and prevent me from using it. Smelter Mark 1 and 3 show up in Production tab however. Smelter Mark 1 and 3 also do not cause an error when using them with the Blueprint system. Regardless if "Use Now" or using the "Saved Blueprint" option is used. 

I've had this issue for several game patches now (minor and any of the major ones). I hope this bug report allows it to get fixed!

Thank you. :)

To Reproduce:
Steps to reproduce the behavior for Production Tab:
1. Have Smelter Mark 2's being manufactured
2. Go to the Production Tab and scroll to the bottom
3. Smelter Mark 2's will not show up after Mark 1's.

To Reproduce:
Steps to reproduce the behavior for Blueprints:
1. Build a Manufacturing setup.
2. Open the Blueprints tab.
3. Use the "Create Blueprint" option to copy the setup.
4. Save the Blueprint (unsure if naming and picking an icon changes the outcome).
5. Exit the Blueprint mode
6. Open the Blueprint and double click to use the setup that was just saved with the Smelter Mark 2 included.
7. Error should appear.

Expected behavior:
No error should appear when using the Smelter Mark 2 in a Blueprint. It should also appear in the Production tab, as the Mark 1 and 3 do already.

Log files:
Unable to upload LogOutput file from BepInEx. It's 39MB and Git only allows up to 25MB. Can provide via an alternative method if necessary.

Screenshots:
![Smelter Mk 2 Bug](https://user-images.githubusercontent.com/2099250/219083140-82c172d8-4153-4eb6-a782-b9a61d275cb3.png)

![Smelter Mk 2 Bug 02](https://user-images.githubusercontent.com/2099250/219083260-e1031909-f585-42f8-9e12-840b51e88eff.png)

![Smelter Mk 2 Bug 03](https://user-images.githubusercontent.com/2099250/219083227-856454db-1152-4c32-a732-5cb03c5de4a8.png)

**Comment by kremnev8** [comment]
Unfortunately I'm unable to reproduce the Blueprint bug. Regarding the production tab bug, I did manage to reproduce it, so I might look into it. However I do think it's minor and so a low priority.
Also please upload the log.

**Comment by kremnev8** [comment]
Ok. I think I fixed the production graphs bug. Update your CommonAPI to get the fix.

**Comment by Flam1ngDem0n** [comment]
How do I further prove that I am having the Blueprint bug?

**Comment by kremnev8** [comment]
Can you please first provide your game log?

**Comment by kremnev8** [comment]
You can also provide the blueprints which cause issues.

**Comment by Flam1ngDem0n** [comment]
I am unable to provide the LogOutput file from BepInEx, as mentioned in the report. It exceeds the Github file upload limit. I will upload a Blueprint instead, since any and all Blueprints that have the Mark 2 Smelter cause the error seen in the screenshot. 

[Magnet.txt](https://github.com/kremnev8/DSP-Mods/files/10785036/Magnet.txt)

**Comment by Flam1ngDem0n** [comment]
[LogOutput.log](https://github.com/kremnev8/DSP-Mods/files/10789573/LogOutput.log)

Here is the log file. Was able to upload it after deleting the previous file.

**Comment by Flam1ngDem0n** [comment]
Hello, is there any solution to the bug that's been found?

**Comment by Flam1ngDem0n** [comment]
Still waiting on any sort of update regarding this problem.

Thank you

**Comment by kremnev8** [comment]
Sorry for taking so long
Looking again at your blueprint I can notice some weird veges instead of smelters. This means your blueprint was created with an earlier version of Better Machines.
Version 1.1.4 should fix this issue.
